server/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -1145,9 +1157,17 @@ public String getDescription() {
         public void writeTo(StreamOutput out) throws IOException {
             out.writeString(targetAllocationID);
             out.writeVLong(primaryTerm);
+            // TODO: Change after backport
+            if (out.getVersion().onOrAfter(Version.V_8_0_0)) {
+                out.writeBoolean(rerouteWasLocal);
+            }
             request.writeTo(out);
         }
 
+        public boolean rerouteWasLocal() {
```

Comment:
The confusing bit about this name is that this flag does not say whether the reroute action came from a local node, but whether the primary action was sent by a reroute phase on the local node.

My suggestion was to mention primaryDelegation here (which is the terminology we use when sending a primary action to a different node (otherwise, it's always local to the node where the reroute was executing). By default primaryDelegation will be false (i.e. exactly the inverse of this flag here)